From 8f838d73daa366683c2118fd542a629d0a67228f Mon Sep 17 00:00:00 2001
From: Michael Dege <michael.dege@renesas.com>
Date: Tue, 29 Nov 2022 21:33:54 +0700
Subject: [PATCH 4/6] arm64: dts: renesas: r8a779f0-s4sk-prototype: Add rswitch
 node

The patch adds rswitch node channel 0/1 for S4 Starter Kit Prototype.
Based on the original (and large) patch by Michael Dege.

Signed-off-by: Michael Dege <michael.dege@renesas.com>
Signed-off-by: Tam Nguyen <tam.nguyen.xa@renesas.com>
---
 .../dts/renesas/r8a779f0-s4sk-prototype.dts   | 69 +++++++++++++++++++
 1 file changed, 69 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/r8a779f0-s4sk-prototype.dts b/arch/arm64/boot/dts/renesas/r8a779f0-s4sk-prototype.dts
index 0bde592b7c38..04d9a2a83368 100644
--- a/arch/arm64/boot/dts/renesas/r8a779f0-s4sk-prototype.dts
+++ b/arch/arm64/boot/dts/renesas/r8a779f0-s4sk-prototype.dts
@@ -16,6 +16,7 @@ / {
 	aliases {
 		serial0 = &hscif0;
 		serial1 = &hscif1;
+		eth0	= &rswitch;
 	};
 
 	chosen {
@@ -167,6 +168,35 @@ pcie1_pins: pcie1 {
 		function = "pcie";
 	};
 
+	tsn0_pins: tsn0 {
+		mux {
+			groups = "tsn0_link", "tsn0_mdio";
+			function = "tsn0";
+			drive-strength = <18>;
+			power-source = <3300>;
+		};
+
+		pins_mdio {
+			groups = "tsn0_mdio";
+			drive-strength = <18>;
+			power-source = <3300>;
+		};
+	};
+
+	tsn1_pins: tsn1 {
+		mux {
+			groups = "tsn1_link", "tsn1_mdio";
+			function = "tsn1";
+			drive-strength = <18>;
+			power-source = <3300>;
+		};
+
+		pins_mdio {
+			groups = "tsn1_mdio";
+			drive-strength = <18>;
+			power-source = <3300>;
+		};
+	};
 };
 
 &rwdt {
@@ -243,3 +273,42 @@ &pciec1_ep {
 	pinctrl-names = "default";
 	clkreq-gpios = <&gpio2 16 GPIO_ACTIVE_LOW>;
 };
+
+&rswitch {
+	status = "okay";
+	pinctrl-0 = <&tsn0_pins &tsn1_pins>;
+	pinctrl-names = "default";
+	#address-cells = <1>;
+	#size-cells = <0>;
+	board-names = "s4sk-prototype";
+
+	ports {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		tsn0: port@0 {
+			reg = <0>;
+			phy-mode = "usxgmii";
+			phy-handle = <&tsn0_phy_2g5>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			tsn0_phy_2g5: ethernet-phy@0 {
+				reg = <0>;
+				compatible = "ethernet-phy-ieee802.3-c45";
+				status = "okay";
+			};
+		};
+
+		tsn1: port@1 {
+			reg = <1>;
+			phy-mode = "usxgmii";
+			phy-handle = <&tsn1_phy_2g5>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			tsn1_phy_2g5: ethernet-phy@1 {
+				reg = <1>;
+				compatible = "ethernet-phy-ieee802.3-c45";
+				status = "okay";
+			};
+		};
+	};
+};
-- 
2.25.1

